core/metacling/test/TClingMethodInfoTests.cxx

```diff
@@ -257,119 +254,146 @@ namespace ROOT11010 {
 
-    EXPECT_EQ(GetInt("baseMemCtorCalled"), 1);
-    EXPECT_EQ(GetInt("baseMemDtorCalled"), 1);
-    EXPECT_EQ(GetInt("baseCtorCalled"), 1);
-    EXPECT_EQ(GetInt("baseDtorCalled"), 1);
+      EXPECT_EQ(GetInt("baseMemCtorCalled"), 1);
+      EXPECT_EQ(GetInt("baseMemDtorCalled"), 1);
+      EXPECT_EQ(GetInt("baseCtorCalled"), 1);
+      EXPECT_EQ(GetInt("baseDtorCalled"), 1);
 
-    EXPECT_EQ(GetInt("derivMemCtorCalled"), 0);
-    EXPECT_EQ(GetInt("derivMemDtorCalled"), 0);
-    EXPECT_EQ(GetInt("derivDtorCalled"), 0);
+      EXPECT_EQ(GetInt("derivMemCtorCalled"), 0);
+      EXPECT_EQ(GetInt("derivMemDtorCalled"), 0);
+      EXPECT_EQ(GetInt("derivDtorCalled"), 0);
 
-    void* argbuf[8];
-    void* objresult = nullptr;
-    std::string s("TheStringCtorArg");
-    argbuf[0] = &s;
-    faceptr.fGeneric(0, 1, argbuf, &objresult);
-    EXPECT_NE(objresult, nullptr);
+      void *argbuf[8];
+      void *objresult = nullptr;
+      std::string s("TheStringCtorArg");
+      argbuf[0] = &s;
+      faceptr.fGeneric(0, 1, argbuf, &objresult);
+      EXPECT_NE(objresult, nullptr);
 
-    EXPECT_EQ(GetInt("baseMemCtorCalled"), 2);
-    EXPECT_EQ(GetInt("baseMemDtorCalled"), 1);
-    EXPECT_EQ(GetInt("baseCtorCalled"), 2);
-    EXPECT_EQ(GetInt("baseDtorCalled"), 1);
+      EXPECT_EQ(GetInt("baseMemCtorCalled"), 2);
+      EXPECT_EQ(GetInt("baseMemDtorCalled"), 1);
+      EXPECT_EQ(GetInt("baseCtorCalled"), 2);
+      EXPECT_EQ(GetInt("baseDtorCalled"), 1);
 
-    EXPECT_EQ(GetInt("derivMemCtorCalled"), 1);
-    EXPECT_EQ(GetInt("derivMemDtorCalled"), 0);
-    EXPECT_EQ(GetInt("derivDtorCalled"), 0);
+      EXPECT_EQ(GetInt("derivMemCtorCalled"), 1);
+      EXPECT_EQ(GetInt("derivMemDtorCalled"), 0);
+      EXPECT_EQ(GetInt("derivDtorCalled"), 0);
 
-    c->Destructor(objresult);
+      c->Destructor(objresult);
 
-    EXPECT_EQ(GetInt("baseMemCtorCalled"), 2);
-    EXPECT_EQ(GetInt("baseMemDtorCalled"), 2);
-    EXPECT_EQ(GetInt("baseCtorCalled"), 2);
-    EXPECT_EQ(GetInt("baseDtorCalled"), 2);
+      EXPECT_EQ(GetInt("baseMemCtorCalled"), 2);
+      EXPECT_EQ(GetInt("baseMemDtorCalled"), 2);
+      EXPECT_EQ(GetInt("baseCtorCalled"), 2);
+      EXPECT_EQ(GetInt("baseDtorCalled"), 2);
 
-    EXPECT_EQ(GetInt("derivMemCtorCalled"), 1);
-    EXPECT_EQ(GetInt("derivMemDtorCalled"), 1);
-    EXPECT_EQ(GetInt("derivDtorCalled"), 1);
+      EXPECT_EQ(GetInt("derivMemCtorCalled"), 1);
+      EXPECT_EQ(GetInt("derivMemDtorCalled"), 1);
+      EXPECT_EQ(GetInt("derivDtorCalled"), 1);
 #endif
-  }
+   }
+}
+
+TEST(TClingMethodInfo, TemplateFun)
+{
+   // PyROOT needs to know whether a ctor is a template, see clingwrapper.cxx:
+   // > // don't give in just yet, but rather get the full name through the symbol name,
+   // > // as eg. constructors do not receive their proper/full name from GetName()
+
+   gInterpreter->Declare(R"CODE(
+struct TemplateFun {
+   template <class T>
+   TemplateFun(int, T);
+
+   template <class T>
+   int NonCtor(float, T);
+};
```

Comment:
```suggestion
};
struct InheritTemplateFun {
    using TemplateFun::TemplateFun;
};
```
and associated testing.

Reply:
That's a very distinct rabbit hole: our `LookupHelper` cannot instantiate function templates, which would cause this test to fail even if the using decl is indeed injecting the base class template. I have added the test skeleton, commented it, and created #6481 #6482